source/slang-capture-replay/slang-module.cpp

```diff
@@ -0,0 +1,234 @@
+#include "capture_utility.h"
+#include "slang-module.h"
+
+namespace SlangCapture
+{
+    ModuleCapture::ModuleCapture(slang::IModule* module)
+        : m_actualModule(module)
+    {
+        assert(m_actualModule != nullptr);
+        slangCaptureLog(LogLevel::Verbose, "%s: %p\n", __PRETTY_FUNCTION__, module);
+    }
+
+    ModuleCapture::~ModuleCapture()
+    {
+        m_actualModule->release();
+    }
+
+    ISlangUnknown* ModuleCapture::getInterface(const Guid& guid)
+    {
+        if(guid == ModuleCapture::getTypeGuid())
+            return static_cast<ISlangUnknown*>(this);
+        else
+            return nullptr;
+    }
+
+    SLANG_NO_THROW SlangResult ModuleCapture::findEntryPointByName(
+        char const*     name,
+        slang::IEntryPoint**   outEntryPoint)
+	{
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+
+        slang::IEntryPoint* pEntryPoint = nullptr;
+        SlangResult res = m_actualModule->findEntryPointByName(name, &pEntryPoint);
+        EntryPointCapture* entryPointCapture = getEntryPointCapture(pEntryPoint);
+        *outEntryPoint = static_cast<slang::IEntryPoint*>(entryPointCapture);
+        return res;
+    }
+
+    SLANG_NO_THROW SlangInt32 ModuleCapture::getDefinedEntryPointCount()
+	{
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+        SlangInt32 res = m_actualModule->getDefinedEntryPointCount();
+        return res;
+    }
+
+    SLANG_NO_THROW SlangResult ModuleCapture::getDefinedEntryPoint(SlangInt32 index, slang::IEntryPoint** outEntryPoint)
+	{
+        // This call is to find the existing entry point, so it has been created already. Therefore, we don't create a new one
+        // and assert the error if it is not found in our map.
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+        slang::IEntryPoint* pEntryPoint = nullptr;
+        SlangResult res = m_actualModule->getDefinedEntryPoint(index, &pEntryPoint);
+
+        if (pEntryPoint)
+        {
+            EntryPointCapture* entryPointCapture = m_mapEntryPointToCapture.tryGetValue(pEntryPoint);
+            if (!entryPointCapture)
+            {
+                assert(!"Entrypoint not found in mapEntryPointToCapture");
+            }
+            *outEntryPoint = static_cast<slang::IEntryPoint*>(entryPointCapture);
+        }
+        else
+            *outEntryPoint = nullptr;
+
+        return res;
+    }
+
+    SLANG_NO_THROW SlangResult ModuleCapture::serialize(ISlangBlob** outSerializedBlob)
+	{
+        slangCaptureLog(LogLevel::Verbose, "%s\n", __PRETTY_FUNCTION__);
+        SlangResult res = m_actualModule->serialize(outSerializedBlob);
+        return res;
+    }
+
+    SLANG_NO_THROW SlangResult ModuleCapture::writeToFile(char const* fileName)
+	{
```

Comment:
The opening braces for the functions looks in a wrong indentation.
There are many other cases.